### ISSUE TITLE
sometimes FFT data is not being passed to histogram

### ISSUE THREAD

**Issue by SerggioC** [matthew-carroll/flutter_ui_challenge_music_player#4]
sample histogram data:
I/flutter: histogram low Size = 15 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
all zeros means no visualizer. 

Have to restart everything to make it work again. 
Maybe something wrong with the audio plugin?

**Comment by Condelab** [comment]
no. ensure you have the record audio permission. both in manifest and during runtime.

**Comment by SerggioC** [comment]
updating to latest  dependency versions I get this error and the visualizer is gone too: 
E/flutter (20030): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: PlatformException(error, Cannot initialize Visualizer engine, error: -3, null)

**Comment by SerggioC** [comment]
> no. ensure you have the record audio permission. both in manifest and during runtime.

Previously the FFT data was updating the visualizer normally but sometimes it would stop, obviously it's not about the permissions.

**Comment by Condelab** [comment]
> 
> 
> > no. ensure you have the record audio permission. both in manifest and during runtime.
> 
> Previously the FFT data was updating the visualizer normally but sometimes it would stop, obviously it was not about the permissions.

i think its the phone or is it simulator? try another phone or simulator!

**Comment by SerggioC** [comment]
Done that. Didn't work. Will check back later.
